tests/unit/button/button_options.js

```diff
@@ -25,135 +25,80 @@ test( "disabled, explicit value", function() {
 	deepEqual( element.prop( "disabled" ), true, "element is not disabled" );
 });
 
-test("disabled, null", function() {
-	expect( 4 );
-	$("#radio01").button({ disabled: null });
-	deepEqual(false, $("#radio01").button("option", "disabled"),
+test( "disabled, null", function() {
+	expect( 2 );
+	$( "#button" ).button({ disabled: null });
+	strictEqual( $("#button").button("option", "disabled"), false,
 		"disabled option set to false");
-	deepEqual(false, $("#radio01").prop("disabled"), "element is disabled");
-
-	$("#radio02").prop("disabled", true).button({ disabled: null });
-	deepEqual(true, $("#radio02").button("option", "disabled"),
-		"disabled option set to true");
-	deepEqual(true, $("#radio02").prop("disabled"), "element is not disabled");
-});
-
-test( "disabled, ui-state-active is removed unless checkbox or radio", function() {
-	expect( 12 );
-	var elements = [
-		$( "<input type='button'>" ),
-		$( "<button></button>" ),
-		$( "<a></a>" ),
-		$( "<div></div>" ),
-		$( "<input type='checkbox' id='checkbox' checked><label for='checkbox'></label>" ),
-		$( "<input type='radio' id='radio' checked><label for='radio'></label>" )
-	];
-
-	$.each( elements, function() {
-		var element = $( this ).first().button(),
-			buttonElement = element.button( "widget" ),
-			elementType = element.prop( "nodeName" ).toLowerCase();
-
-		if ( element.is( "input" ) ) {
-			elementType += ":" + element.attr( "type" );
-		}
-
-		element.trigger( "mousedown" );
-		ok( buttonElement.hasClass( "ui-state-active" ),
-			"[" + elementType + "] has ui-state-active class after mousedown." );
-
-		element.button( "disable" );
-		if ( element.is( "[type=checkbox], [type=radio]" ) ) {
-			ok( buttonElement.hasClass( "ui-state-active" ),
-				"Disabled [" + elementType + "] has ui-state-active class." );
-		} else {
-			ok( !buttonElement.hasClass( "ui-state-active" ),
-				"Disabled [" + elementType + "] does not have ui-state-active class." );
-		}
-	});
+	strictEqual( $("#button").prop("disabled"), false, "element is disabled");
 });
 
-test("text false without icon", function() {
+test( "showLabel false without icon", function() {
 	expect( 1 );
-	$("#button").button({
-		text: false
+	$( "#button" ).button({
+		showLabel: false
 	});
-	ok( $("#button").is(".ui-button-text-only:not(.ui-button-icon-only)") );
+	strictEqual( $( "#button" ).is( ":ui-button.ui-corner-all.ui-widget" ), true,
```

Comment:
We can make this more precise:

```js
strictEqual( $( "#button" ).attr("class"), "ui-button ui-corner-all ui-widget",
		"Button has correct classes" );
```

Scott has argued against this style of asserting classes, I still think its superior, since it catches any unwanted additional classes. And the order of the classes is deterministic. If the source code changes, it may be necessary to update assertions like this, but I think that's worth it.